py/bitbox02/communication/bitbox_api_protocol.py

```diff
@@ -197,26 +215,43 @@ def __init__(
                     self.close()
                     raise Exception("Unlock process aborted")
 
+        self.noise = self._create_noise_channel(show_pairing_callback, noise_config)
+
+    # pylint: disable=too-many-branches
+    def _create_noise_channel(
+        self,
+        show_pairing_callback: Callable[[str], bool],
+        noise_config: Optional[BitBoxNoiseConfig] = None,
+    ) -> NoiseConnection:
         if self._query(OP_I_CAN_HAS_HANDSHAEK) != RESPONSE_SUCCESS:
             self.close()
             raise Exception("Couldn't kick off handshake")
 
         # init noise channel
         noise = NoiseConnection.from_name(b"Noise_XX_25519_ChaChaPoly_SHA256")
         noise.set_as_initiator()
-        dummy_private_key = os.urandom(32)
-        noise.set_keypair_from_private_bytes(Keypair.STATIC, dummy_private_key)
+        if noise_config is not None:
```

Comment:
if it is none there is no private_key?

Reply:
Whoops, the private_key is now always generated, but not used if the config provides one.